rfcs/0130-Implementing Checks API in tc-github while preserving Statuses API

```diff
@@ -0,0 +1,44 @@
+# RFC 130 - Implementing Checks API
+* Comments: [#130](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/130)
+* Proposed by: @djmitche
+
+# Summary
+
+We want to implement Checks API without dropping the support os Statuses API. Statuses API would be the default. Our users
+will have the freedom to choose which API they want by indicating their choice in `.taskcluster.yml`.
+
+## Motivation
+
+So GitHub has come up with a new API to create status indicators in its UI, called Checks.
+Checks API isn't in any way compatible with the current Statuses API - status created with one cannot be updated with another;
+they are two totally separate sets of data. Calling both APIs would result in two sets of status indicators in the UI.
+Any third party tools and integrations that rely on Statuses API to work would stop working after we switched to Checks API.
+
+We want to implement Checks because it seems to be the future, and because it gives a much richer output and a whole bunch 
+of controls for our users.
+
+At the same time, we want to preserve compatibility with Statuses because some of our users use tools and integration that 
+rely on them.
+
+# Details
+
+Currently we create a task and a status in a single function that listens
+to github webhooks. The idea is to create a task in separate function that listens to webhooks;
+then listen to the pulse message (`task defined`? `artifact created`?) on a certain queue that signals about creation of a task.
+The messages would go to this or that queue based on some indication in `taskcluster.yml` - basically, "use Checks API" or the
+absence of "use Checks API". The presence of thet indication would result in adding a route (like a path to a queue?)
+to the task definition.
```

Comment:
Possibly could also keep track of which trask groups are created and listen for [taskGroupResolved](https://docs.taskcluster.net/docs/reference/platform/taskcluster-queue/references/events#taskGroupResolved) to decide when to set the outcome, either with checks or status api.

Reply:
Oh that's great stuff for Statuses! Thank you, I'll try it out. Checks work a bit differently: there's a concept of a Check-suite, which is a set of checkruns, and it resolves by Github automatically based on the resolutions of the checkruns. So we don't need to track group statuses for Checks ourselves

Reply:
the group thing is so that we could dependencies in task graphs and only update the check's outcome when all of the dependencies resolve.  Like, if we have builds and tests, we might only want to update when the tests are done, rather than having an update for every single task.

Reply:
A lot of my thinking is here: https://github.com/taskcluster/taskcluster-rfcs/pull/131#discussion_r230321814 .  But basically, it seems from my reading of the checks docs and examples that it should be a 1:1 mapping between tool and check.  In the case where we have a statically defined set of tasks with no dependencies, this mapping of task to check isn't a problem.  It likely will become a problem when we get dependencies and decision tasks in .taskcluster.yml.

Reply:
The taskcluster-web repository has some integrations installed that post a few checkruns for each revision (so 1:N mapping).  For tasks, we should be able to support zero, one, or many checkRuns per task.

Reply:
How decision tasks and dependencies are indicated in the `.taskcluster.yml`?

Reply:
>a few checkruns

You probably mean check-suites. Yeah, that would be more convenient, wouldn't - each decision task would get its own check-suite. I wonder now, how can tc-gh be made aware of the difference between decision tasks and regular tasks

Reply:
oh, I see there's `decision_task_id` in the yml

Reply:
I'm sorta thinking that we can do `checks-v1` without having anything fancy to do with 1:N mappings and then take on a `checks-v2` later? In the first version each task would get a checkrun or something like that.

Reply:
certainly! 

In fact, if we are going to proceed with this idea here, I would first implement this way of creating statuses (create a task on github webhook, post a status on pulse message), and then add checks in there, in their simplest form possible, hidden from users